curve25519-dalek/src/backend/serial/u64/field_lemmas/limbs_to_bytes_lemmas.rs

```diff
@@ -0,0 +1,2396 @@
+    assert(bytes[4] as nat == (limb0_low48 as nat / pow2(32)) % 256);
+    assert(bytes[5] as nat == (limb0_low48 as nat / pow2(40)) % 256);
+
+    // Now apply our reconstruction lemma
+    lemma_sum_extracted_bytes_reconstructs_value(
+        limb0_low48,
+        bytes[0],
+        bytes[1],
+        bytes[2],
+        bytes[3],
+        bytes[4],
+        bytes[5],
+        6,
+    );
+
+    // The reconstruction lemma tells us: low_48_bits == limbs[0] % 2^48
+    assert(low_48_bits == (limbs[0] as nat % pow2(48)));
+
+    // Step 2: Show the contribution from byte 6
+    // From bytes_match_limbs_packing: bytes[6] == ((limbs[0] >> 48) | (limbs[1] << 3)) as u8
+    // The low 3 bits of bytes[6] come from limbs[0] >> 48
+    // which is limbs[0] / 2^48
+
+    // Since limbs[0] < 2^51, we have limbs[0] / 2^48 < 2^3 = 8
+    assert(limbs[0] < pow2(51));
+    lemma_div_bound(limbs[0] as nat, 48, 51);
+    assert(limbs[0] as nat / pow2(48) < pow2((51 - 48) as nat));
+    assert(limbs[0] as nat / pow2(48) < pow2(3));
+    lemma2_to64();
+    assert(pow2(3) == 8);
+    assert(limbs[0] as nat / pow2(48) < 8);
+
+    // The high 5 bits of byte 6 come from limbs[1], so the low 3 bits are:
+    let high_3_bits_contribution = ((limbs[0] as nat / pow2(48)) % 8) * pow2(6 * 8);
+
+    // Since limbs[0]/2^48 < 8, taking % 8 is identity
+    assert((limbs[0] as nat / pow2(48)) % 8 == limbs[0] as nat / pow2(48));
+    assert(high_3_bits_contribution == (limbs[0] as nat / pow2(48)) * pow2(48));
+
+    // Step 3: Apply div-mod theorem
+    // limbs[0] = (limbs[0] % 2^48) + (limbs[0] / 2^48) * 2^48
+    lemma_pow2_pos(48);  // Establishes pow2(48) > 0
+    assert(pow2(48) > 0);
+    lemma_fundamental_div_mod(limbs[0] as int, pow2(48) as int);
+    assert(limbs[0] as nat == (limbs[0] as nat % pow2(48)) + (limbs[0] as nat / pow2(48)) * pow2(
+        48,
+    ));
+
+    // Step 4: Show this equals limb0_byte_contribution
+    assert(limb0_byte_contribution(limbs, bytes) == low_48_bits + high_3_bits_contribution);
+    assert(limb0_byte_contribution(limbs, bytes) == (limbs[0] as nat % pow2(48)) + (limbs[0] as nat
+        / pow2(48)) * pow2(48));
+    assert(limb0_byte_contribution(limbs, bytes) == limbs[0] as nat);
+}
+
+/// Helper: 5-byte reconstruction lemma
+/// Proves that 5 consecutive bytes reconstruct a 40-bit value
+proof fn lemma_5_bytes_reconstruct(
+    value: nat,
+    byte0: u8,
+    byte1: u8,
+    byte2: u8,
+    byte3: u8,
+    byte4: u8,
+)
+    requires
+        byte0 as nat == (value / pow2(0)) % 256,
+        byte1 as nat == (value / pow2(8)) % 256,
+        byte2 as nat == (value / pow2(16)) % 256,
+        byte3 as nat == (value / pow2(24)) % 256,
+        byte4 as nat == (value / pow2(32)) % 256,
+        value < pow2(40),  // 5 bytes = 40 bits
+
+    ensures
+        byte0 as nat * pow2(0) + byte1 as nat * pow2(8) + byte2 as nat * pow2(16) + byte3 as nat
+            * pow2(24) + byte4 as nat * pow2(32) == value,
```

Comment:
This seems like conceptual duplication with `lemma_sum_extracted_bytes_reconstructs_value`